L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc

```diff
@@ -0,0 +1,233 @@
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.h"
+#include "DataFormats/MuonDetId/interface/CSCTriggerNumbering.h"
+
+CSCGEMMotherboard::CSCGEMMotherboard(unsigned endcap, unsigned station,
+                                     unsigned sector, unsigned subsector,
+                                     unsigned chamber,
+                                     const edm::ParameterSet& conf) :
+  CSCUpgradeMotherboard(endcap, station, sector, subsector, chamber, conf)
+  , maxDeltaBXPad_(tmbParams_.getParameter<int>("maxDeltaBXPad"))
+  , maxDeltaBXCoPad_(tmbParams_.getParameter<int>("maxDeltaBXCoPad"))
+  , useOldLCTDataFormat_(tmbParams_.getParameter<bool>("useOldLCTDataFormat"))
+  , promoteALCTGEMpattern_(tmbParams_.getParameter<bool>("promoteALCTGEMpattern"))
+  , promoteALCTGEMquality_(tmbParams_.getParameter<bool>("promoteALCTGEMquality"))
+  , doLCTGhostBustingWithGEMs_(tmbParams_.getParameter<bool>("doLCTGhostBustingWithGEMs"))
+{
+  gemId = GEMDetId(theRegion, 1, theStation, 1, theChamber, 0).rawId();
+  
+  const edm::ParameterSet coPadParams(conf.getParameter<edm::ParameterSet>("copadParam"));
+  coPadProcessor.reset( new GEMCoPadProcessor(endcap, station, 1, chamber, coPadParams) );
+
+  maxDeltaPadL1_ = (par ? tmbParams_.getParameter<int>("maxDeltaPadL1Even") :
+		    tmbParams_.getParameter<int>("maxDeltaPadL1Odd") );
+  maxDeltaPadL2_ = (par ? tmbParams_.getParameter<int>("maxDeltaPadL2Even") :
+		    tmbParams_.getParameter<int>("maxDeltaPadL2Odd") );
+}
+
+CSCGEMMotherboard::CSCGEMMotherboard() : CSCUpgradeMotherboard()
+{
+}
+
+CSCGEMMotherboard::~CSCGEMMotherboard()
+{
+}
+
+void CSCGEMMotherboard::clear()
+{
+  pads_.clear();
+  coPads_.clear();    
+}  
+
+
+void CSCGEMMotherboard::run(const CSCWireDigiCollection* wiredc,
+			    const CSCComparatorDigiCollection* compdc,
+			    const GEMPadDigiClusterCollection* gemClusters)
+{
+  std::unique_ptr<GEMPadDigiCollection> gemPads(new GEMPadDigiCollection());
+  coPadProcessor->declusterize(gemClusters, *gemPads);
+  run(wiredc, compdc, gemPads.get());
+}
+
+
+void CSCGEMMotherboard::retrieveGEMPads(const GEMPadDigiCollection* gemPads, unsigned id)
+{
+  auto superChamber(gem_g->superChamber(id));
+  for (const auto& ch : superChamber->chambers()) {
+    for (const auto& roll : ch->etaPartitions()) {
+      GEMDetId roll_id(roll->id());
+      auto pads_in_det = gemPads->get(roll_id);
+      for (auto pad = pads_in_det.first; pad != pads_in_det.second; ++pad) {
+        auto id_pad = std::make_pair(roll_id, *pad);
+        const int bx_shifted(lct_central_bx + pad->bx());
+        for (int bx = bx_shifted - maxDeltaBXPad_;bx <= bx_shifted + maxDeltaBXPad_; ++bx) {
+          pads_[bx].push_back(id_pad);  
```

Comment:
Hi @dildick - I think I finally got through this PR now - sorry for the delay

I guess this function isn't yet used - but it would seems to need call to clear() to be safe?